test/controllers/sections/items/ItemProducerSizeControllerSpec.scala

```diff
@@ -159,7 +170,59 @@ class ItemProducerSizeControllerSpec extends SpecBase with MockPreDraftService w
         status(result) mustEqual SEE_OTHER
         redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
       }
+    }
+
+    "for a GET unableToProvideInformation" - {
+
+      "must redirect to the next page" in new Setup(Some(
+        emptyUserAnswers.set(ItemExciseProductCodePage(0), testEpcWine)
+      )) {
+
+        MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
+
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.unableToProvideInformation(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual testOnwardRoute.url
+      }
+
+      "must redirect to Index of section when the idx is outside of bounds" in new Setup(Some(
+        emptyUserAnswers.set(ItemExciseProductCodePage(0), testEpcWine)
+      )) {
+
+        MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
+
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(GET, itemProducerSizeRoute(1))
+
+        val result = TestController.unableToProvideInformation(testErn, testDraftId, 1, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual testOnwardRoute.url
+      }
```

Comment:
I don't think this test is validating the index out of bounds logic, that logic doesn't seem to be applied on this method in the controller

Reply:
This still needs to be addressed